packages/pds/src/services/actor/views.ts

```diff
@@ -0,0 +1,106 @@
+
+    const { ref } = this.db.db.dynamic
+
+    const profileInfos = await this.db.db
+      .selectFrom('did_handle')
+      .where(
+        'did_handle.did',
+        'in',
+        results.map((r) => r.did),
+      )
+      .leftJoin('profile', 'profile.creator', 'did_handle.did')
+      .select([
+        'did_handle.did as did',
+        'profile.uri as profileUri',
+        'profile.displayName as displayName',
+        'profile.description as description',
+        'profile.avatarCid as avatarCid',
+        'profile.bannerCid as bannerCid',
+        this.db.db
+          .selectFrom('follow')
+          .whereRef('creator', '=', ref('did_handle.did'))
+          .select(countAll.as('count'))
+          .as('followsCount'),
+        this.db.db
+          .selectFrom('follow')
+          .whereRef('subjectDid', '=', ref('did_handle.did'))
+          .select(countAll.as('count'))
+          .as('followersCount'),
+        this.db.db
+          .selectFrom('post')
+          .whereRef('creator', '=', ref('did_handle.did'))
+          .select(countAll.as('count'))
+          .as('postsCount'),
+        this.db.db
+          .selectFrom('follow')
+          .where('creator', '=', viewer)
+          .whereRef('subjectDid', '=', ref('did_handle.did'))
+          .select('uri')
+          .as('requesterFollow'),
+        this.db.db
+          .selectFrom('mute')
+          .whereRef('did', '=', ref('did_handle.did'))
+          .where('mutedByDid', '=', viewer)
+          .select('did')
+          .as('requesterMuted'),
+      ])
+      .execute()
+
+    const profileInfoByDid = profileInfos.reduce((acc, info) => {
+      return Object.assign(acc, { [info.did]: info })
+    }, {} as Record<string, ArrayEl<typeof profileInfos>>)
+
+    const views = results.map((result) => {
+      const profileInfo = profileInfoByDid[result.did]
+      const avatar = profileInfo?.avatarCid
+        ? this.imgUriBuilder.getCommonSignedUri('avatar', profileInfo.avatarCid)
+        : undefined
+      const banner = profileInfo?.bannerCid
+        ? this.imgUriBuilder.getCommonSignedUri('banner', profileInfo.bannerCid)
+        : undefined
+      return {
+        did: result.did,
+        declaration: getDeclarationSimple(result),
+        handle: result.handle,
+        creator: result.did,
+        displayName: profileInfo?.displayName || undefined,
+        description: profileInfo?.description || undefined,
+        avatar,
+        banner,
+        followsCount: profileInfo?.followsCount ?? 0,
+        followersCount: profileInfo?.followersCount ?? 0,
+        postsCount: profileInfo?.postsCount ?? 0,
+        myState: {
+          follow: profileInfo?.requesterFollow || undefined,
+          muted: !!profileInfo?.requesterMuted,
+        },
+      }
+    })
+
+    return Array.isArray(result) ? views : views[0]
+  }
+}
+
+type UserResult = User & DidHandle & RepoRoot
+type ArrayEl<A> = A extends readonly (infer T)[] ? T : never
```

Comment:
haha cheeky

we may have to put this in `common` or something, seems nifty

Reply:
Good call, done!  It was being used in a couple places, so seems like the right move.